src/hotspot/share/gc/shared/gcVMOperations.hpp

```diff
@@ -150,10 +144,11 @@ class VM_GC_Operation: public VM_GC_Sync_Operation {
 
   // Acquire the reference synchronization lock
   virtual bool doit_prologue();
```

Comment:
This does a lot more than just acquiring the lock.  It also handles the prevention of multiple gc requests.